q2_sample_classifier/plugin_setup.py

```diff
@@ -446,17 +448,19 @@
 )
 
 
+T = TypeMatch([Frequency, RelativeFrequency, PresenceAbsence, Balance,
+              PercentileNormalized])
 plugin.methods.register_function(
     function=split_table,
-    inputs=inputs,
+    inputs={'table': FeatureTable[T]},
     parameters={
         'random_state': parameters['base']['random_state'],
         'missing_samples': parameters['base']['missing_samples'],
         **parameters['splitter'],
         'metadata': MetadataColumn[Numeric | Categorical],
         **parameters['regressor']},
-    outputs=[('training_table', FeatureTable[Frequency]),
-             ('test_table', FeatureTable[Frequency])],
+    outputs=[('training_table', FeatureTable[T]),
+             ('test_table', FeatureTable[T])],
     input_descriptions=input_descriptions,
```

Comment:
Let's branch the description, too, that way it doesn't remain coupled to the old global `inputs`:

```suggestion
     input_descriptions={'table': 'Feature table containing all features that should be used for target prediction.'},
```